fetch.js

```diff
@@ -311,7 +311,13 @@
       xhr.onload = function() {
         var status = (xhr.status === 1223) ? 204 : xhr.status
         if (status < 100 || status > 599) {
```

Comment:
The `onload` event must be fired when "[the request has successfully completed](http://www.w3.org/TR/XMLHttpRequest/#event-xhr-load)", which means there is an HTTP status code available in the response from the server. If there is no response, as in the case of a network failure, `onerror` must be triggered.

Here's [the commit](https://github.com/github/fetch/commit/5a2d5f860b399f6d4dcac302788f77d594b6adc1) describing the current status check for IE 9.

I'm not familiar with React Native, but this might be a bug in the implementation of XHR. The [supported browsers](https://github.com/github/fetch#browser-support) are listed in the readme.